src/Core/Kernel.php

```diff
@@ -139,7 +140,11 @@ public static function getConnection(): Connection
 
     public function getCacheDir(): string
     {
-        $pluginHash = md5(implode('', array_keys($this->pluginLoader->getPluginInstances()->getActives())));
+        $pluginFeatures = array_map(function (Plugin $plugin): string {
```

Comment:
I'd rather see the version of the plugin here. Just because a plugin was updated in the database doesn't mean that a new code state was deployed.

For development purposes nice, but for live systems rather vulnerable.

Reply:
Good point. Plugin version is not bad at all. Also better available in a more static way as it can be read from composer.json.